catalyst_voices/packages/internal/catalyst_voices_repositories/lib/src/document/source/database_drafts_data_source.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:catalyst_voices_repositories/catalyst_voices_repositories.dart';
+
+/// Encryption will be added later here as drafts are not public
+final class DatabaseDraftsDataSource implements DraftDataSource {
+  final CatalystDatabase _database;
+
+  DatabaseDraftsDataSource(
+    this._database,
+  );
+
+  @override
+  Future<bool> exists({required DocumentRef ref}) {
+    return _database.draftsDao.count(ref: ref).then((count) => count > 0);
+  }
+
+  @override
+  Future<DocumentData> get({required DocumentRef ref}) async {
+    final entity = await _database.draftsDao.query(ref: ref);
+    if (entity == null) {
+      throw DraftNotFound(ref: ref);
+    }
+
+    return entity.toModel();
+  }
+
+  @override
+  Stream<DocumentData?> watch({required DocumentRef ref}) {
+    return _database.draftsDao
+        .watch(ref: ref)
+        .map((entity) => entity?.toModel());
+  }
+
+  @override
+  Future<void> save({required DocumentData data}) async {
+    final idHiLo = UuidHiLo.from(data.metadata.id);
+    final verHiLo = UuidHiLo.from(data.metadata.version);
+
+    final draft = DraftEntity(
+      idHi: idHiLo.high,
+      idLo: idHiLo.low,
+      verHi: verHiLo.high,
+      verLo: verHiLo.low,
+      type: data.metadata.type,
+      content: data.content,
+      metadata: data.metadata,
+      title: data.content.title ?? '',
+    );
+
+    await _database.draftsDao.save(draft);
+  }
+
+  @override
+  Future<void> update({
+    required DocumentRef ref,
+    required DocumentDataContent content,
+  }) async {
+    await _database.draftsDao.updateContent(ref: ref, content: content);
+  }
+}
+
+extension _DraftEntityMapper on DraftEntity {
```

Comment:
You can declare it like this, then it will be private (in this file):

```suggestion
extension on DraftEntity {
```

Reply:
Awesome! Didn't know that